server/api/api.go

```diff
@@ -100,26 +121,44 @@ func (api *WsApi) ProcessRawRequest(context context.Context, messageType int, me
 		return nil, fmt.Errorf("invalid request JSON format")
 	}
 
+	suid := context.Value("suid").(uuid.UUID).String()
+	log.Info().Msgf("WS started '%s' request from suid: %s, req: %s", messageObj.Request, suid, messageObj.Payload)
+
 	// get user info from context
 	user := context.Value("user").(*models.User)
 
 	if handler, found := handlersMap[messageObj.Request]; found {
 		if handler.messageType != messageType {
+			log.Info().Msgf("WS request from: %s has wrong message type: %d", suid, messageType)
 			return nil, fmt.Errorf("message type is wrong")
 		}
+
 		if handler.needAuth && user == nil {
-			return nil, fmt.Errorf("user unauthorized")
+			log.Info().Msgf("WS request from: %s unauthorized", suid)
+			return respondWithError(messageObj.Id, interfaces.UnauthorizedError), nil
 		}
 
-		log.Debug().Msgf("Started '%s' request from suid: %s", messageObj.Request, context.Value("suid").(uuid.UUID).String())
-
-		return handler.handler(context, &interfaces.ApiRequest{
+		// process request
+		wsResp, handlerError := handler.handler(context, &interfaces.ApiRequest{
 			UseCases: api.useCases,
 			Repos:    api.repos,
 			User:     user,
 			Data:     &messageObj,
 		})
+
+		if handlerError != nil {
+			if handlerError.Code == interfaces.InternalError {
```

Comment:
maybe implement `String()` method for `HandlerError` ?

Reply:
why? what profit?

Reply:
you could just write `handlerError.Error()` instead of a large `if else block`

Reply:
hmm, unfortunately,  It won't help me(
Anyway, I need if else block to handle different errors type, in case of internal I log with error level, in another case use just debug one